### PR TITLE
[Dubbo-6641]add a redis config center module

### LINKED ISSUE #6641: Add a dubbo-configcenter-redis module

### PR DESCRIPTION
**Pull request by tswstarplanet**
## What is the purpose of the change

fix #6641

## Brief changelog

1.Add a redis config center module
2.Add some method to RedisClient interface.

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-configcenter/dubbo-configcenter-redis/src/main/java/org/apache/dubbo/configcenter/support/redis/RedisDynamicConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    private void checkConfig() {
+        for (Map.Entry<String, HashHolder> entry : configHashes.entrySet()) {
+            String value = redisClient.get(entry.getKey());
+            if (value != null) {
+                Integer hash = Integer.valueOf(value.hashCode());
+                Integer originHash = entry.getValue().getHash();
+                if (!hash.equals(originHash)) {
+                    System.out.println("-------------key: " + entry.getKey() + ", -----------value: " + entry.getValue());
```
Comment (AlbumenJ) [inline dubbo-configcenter/dubbo-configcenter-redis/src/main/java/org/apache/dubbo/configcenter/support/redis/RedisDynamicConfiguration.java:148]: I think using `System.io` to output is need to banned. pls use `logger` instead.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-configcenter/dubbo-configcenter-redis/src/main/java/org/apache/dubbo/configcenter/support/redis/RedisDynamicConfiguration.java (hunk 1 above)
Comment (tswstarplanet) [reply dubbo-configcenter/dubbo-configcenter-redis/src/main/java/org/apache/dubbo/configcenter/support/redis/RedisDynamicConfiguration.java:148]: Thanks! It is my test code, and I forgot to delete.

**Comment by chickenlj** [comment]
@tswstarplanet Would you mind re-submit this PR to apache/spi-dubbo-extensions？

**Comment by chickenlj** [comment]
apache/dubbo-spi-extensions

**Comment by tswstarplanet** [comment]
OK

**Comment by chickenlj** [comment]
Let's track this extension on apache/dubbo-spi-extensions

**Pull request closed without merging**